### PR TITLE
Support for S3_RGW (CephFS S3-compatible API)

### PR DESCRIPTION
**Pull request by nicholasamorim**
Hi, this is a proposal to add S3_RGW to the list of supported providers.

I'm actually not sure how to add tests - or even if I need to so. I have been using in our setup and it's working fine.

Aims to solve #31

### CODE REVIEW

**Comment by nicholasamorim** [comment]
@adejanovski 

- fixed the hash comparison
- added `is_aws_s3`
- updated readme

File: medusa/storage/s3_rgw.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    def connect_storage(self):
+        with io.open(os.path.expanduser(self.config.key_file), 'r', encoding='utf-8') as json_fi:
+            credentials = json.load(json_fi)
+
+        driver = S3RGWStorageDriver(
+            host=self.config.host,
+            port=self.config.port,
+            region="default",
+            signature_version="4",
+            key=credentials['aws_access_key_id'],
```
Comment (adejanovski) [inline medusa/storage/s3_rgw.py]: It seems a bit confusing to use a credential file that mentions AWS although we're not using an AWS account (unless I'm missing something).
Is it a standard way of authenticating to CEPH using the S3 api? Are aws credentials involved in any way?

Comment (nicholasamorim) [reply medusa/storage/s3_rgw.py]: Good point, they strive to keep the nomenclature the same but I clearly forgot to remove the `aws_`. Done.

**Comment by nicholasamorim** [comment]
@adejanovski done

**Pull request merged**